storm-client/src/jvm/org/apache/storm/messaging/netty/StormClientHandler.java

```diff
@@ -47,12 +47,24 @@ public void channelRead(ChannelHandlerContext ctx, Object message) throws Except
             BackPressureStatus status = (BackPressureStatus) message;
             if (status.bpTasks != null) {
                 for (Integer bpTask : status.bpTasks) {
-                    remoteBpStatus[bpTask].set(true);
+                    try {
+                        remoteBpStatus[bpTask].set(true);
+                    } catch (ArrayIndexOutOfBoundsException e) {
+                        //Just in case we get something we are confused about
+                        // we can continue processing the rest of the tasks
+                        LOG.error("BP index out of bounds {}", e);
```

Comment:
Is it intentional that the stack trace is stripped here?

Reply:
Ah! I realized `e` is bound to {} so my suggestion is already applied (show index within same line), but no stack trace.

Reply:
Yes, because when this was happening it was happening everywhere, and there is only one line under the try catch block so there isn't a lot of confusion about what happened.